citrination_client/models/client.py

```diff
@@ -63,16 +66,21 @@ def predict(self, data_view_id, candidates, method="scalar", use_prior=True):
         :return: The results of the prediction
         :rtype: list of :class:`PredictionResult`
         """
-        body = self._get_predict_body(candidates, method, use_prior)
-        failure_message = "Error while making prediction for data view {}".format(data_view_id)
-        response_dict = self._get_success_json(
-            self._post_json(routes.data_view_predict(data_view_id), data=body, failure_message=failure_message))
-        candidate_dicts = response_dict["candidates"]
-        return list(
-            map(
-                lambda c: _get_prediction_result_from_candidate(c), candidate_dicts
+
+        uid = self.submit_predict_request(data_view_id, candidates, method, use_prior)
+
+        while self.check_predict_status(data_view_id, uid)['status'] == "running":
```

Comment:
Isn't this the endpoint where we settled on [Failed, Killed, Accepted, Finished, Processing] as statuses?

Reply:
You're commenting on a past commit (this was fixed in 3a9c464)

Reply:
ah ok, I was reviewing commit by commit